system-contracts/contracts/Compressor.sol

```diff
@@ -37,34 +38,62 @@ contract Compressor is ICompressor, ISystemContract {
     ///         * If the chunk is not already in the dictionary, it is added to the dictionary array.
     ///         * If the dictionary becomes overcrowded (2^16 + 1 elements), the compression process will fail.
     ///         * The 2-byte index of the chunk in the dictionary is added to the encoded data.
+    /// @dev Note: The functions reverts the execution if:
+    ///     - The bytecode has non expected format:
+    ///         * Bytecode bytes length is not a multiple of 32.
+    ///         * Bytecode bytes length is not less than 2^21 bytes (2^16 words).
+    ///         * Bytecode words length is not odd.
+    ///     - The encoded data is incorrect:
+    ///         * The encoded data bytes length is not 4 times shorter than the original bytecode bytes length.
+    ///         * There are more dictionary entries than encoded data entries.
+    ///         * The encoded data contains an index that is out of bounds of the dictionary.
+    ///         * Any encoded chunk does not match the original bytecode
+    ///         * The dictionary contains unused entries.
     /// @dev Currently, the method may be called only from the bootloader because the server is not ready to publish bytecodes
     /// in internal transactions. However, in the future, we will allow everyone to publish compressed bytecodes.
+    /// @dev Read more about the compression: https://github.com/matter-labs/zksync-era/blob/main/docs/guides/advanced/compression.md
     function publishCompressedBytecode(
         bytes calldata _bytecode,
         bytes calldata _rawCompressedData
     ) external payable onlyCallFromBootloader returns (bytes32 bytecodeHash) {
         unchecked {
+            // Validate and hash the bytecode, ensuring its length is a multiple of 32 bytes
+            bytecodeHash = Utils.hashL2Bytecode(_bytecode);
+
+            // The dictionary length is a multiple of 8
             (bytes calldata dictionary, bytes calldata encodedData) = _decodeRawBytecode(_rawCompressedData);
 
-            require(dictionary.length % 8 == 0, "Dictionary length should be a multiple of 8");
-            require(dictionary.length <= 2 ** 16 * 8, "Dictionary is too big");
+            // This also ensures that the encodedData length is a multiple of 2
             require(
-                encodedData.length * 4 == _bytecode.length,
+                _bytecode.length / 4 == encodedData.length,
```

Comment:
Why? `encodedData.length * 4 == _bytecode.length` is stronger comparison then `_bytecode.length / 4 == encodedData.length` as theoretically `_bytecode.length` may not be dividable by 4 (I know it is enforced in `hashL2Bytecode`), but reason of changing is not clear to me

Reply:
It is indeed "stronger", but if the dividability is ensured, there is no reason to doubt that. In my opinion this is a readability improvement.
If our biggest priority is to minimise changes, then I can revert back my changes.

Reply:
It is very subjective of "readability improvement", keeping diff small is the priority - yes. Please revert unneeded changes